poll/poll_test.go

```diff
@@ -38,6 +40,52 @@ func TestWaitOn(t *testing.T) {
 	assert.Equal(t, end, counter)
 }
 
+func TestWaitOnFile(t *testing.T) {
+	fakeT := &fakeT{}
+	fakeFilePath := "./fakefile"
+	defer os.Remove(fakeFilePath)
+
+	go func() {
+		time.Sleep(time.Second)
```

Comment:
I always try to avoid `time.Sleep` in tests, because it slows the test down. If it's necessary a shorter sleep of 100ms might be ok, but I think we can avoid it in this case.

Another way to test this is to call the function directly without using `WaitOn`:

1. call the function once, assert it returned continue
2. create the file
3. call the function again, assert it returned success

Each of these 2 cases could be wrapped in a `t.Run()` as well, to give more context about the failure.
I think with this approach you won't need to "sanity check" type assertions that exist now, which I find makes the test read a bit better. Sanity assertions tend to distract from the behaviour that is being asserted.

I'm not sure what errors stat can return, so not sure if we can test the error case.

Reply:
Yes I agree with you. I was not quiet happy with this code. Your solution looks better. Also, as we don't use `WaitOn` anymore, I think I'm going to move those test in a `check_test.go` file. What do you think about that?

Reply:
Good idea!